core/src/main/java/org/apache/iceberg/deletes/SortingPositionOnlyDeleteWriter.java

```diff
@@ -60,7 +72,7 @@ public void write(PositionDelete<T> positionDelete) {
 
   @Override
   public long length() {
-    return writer.length();
+    return result != null ? result.totalFileSizeInBytes() : 0L;
```

Comment:
Can you note this behavior in the Javadoc? I think it is correct to only produce the size once it has been closed an produces a result, since that would avoid any problem from wrapping this in a `RollingFileWriter`. But it is still unexpected that this isn't accurate during the write.

Reply:
I'll probably switch to not implementing it at all, just like we do in the other writer.